src/components/ContactPolicy/ContactPolicy.test.js

```diff
@@ -77,16 +82,19 @@ describe('ContactPolicy component', () => {
   });
 
   it('should show success message when updating data correctly', async () => {
-    // // Act
-    const { getByText } = render(<ContactPolicyComponent />);
+    // Act
+    const { container, findByText, getByText } = render(<ContactPolicyComponent />);
 
-    //TODO: Improve this code
-    await waitFor(async () => {
-      fireEvent.click(getByText('common.save'));
+    // Assert
+    const loadingBox = container.querySelector('.wrapper-loading');
+    await waitForElementToBeRemoved(loadingBox);
 
-      // Assert
-      expect(getByText('contact_policy.success_msg'));
-    });
+    const switchButton = container.querySelector('input#contact-policy-switch');
+    fireEvent.click(switchButton);
+    expect(switchButton).toBeChecked();
+
+    fireEvent.click(getByText('common.save'));
+    expect(await findByText('contact_policy.success_msg'));
   });
 
   it('should disable the inputs if the switch is not active', async () => {
```

Comment:
The test says inputs should be disabled, but I don't see this being verified

Reply:
The switch is disabled by default, therefore it is verified that the inputs are disabled

Reply:
I can see is that it is verifying that once the switch is clicked, the inputs are enabled. But you also have to make the checks that before clicking, the inputs are disabled

Reply:
You mean that in this test `"shouldn't disable the inputs if the switch is active"`, right?
Yes, it would be good to include that previous check.

Reply:
Yes, it's better